app/src/main/java/com/example/sam/bucketlist/api/APIManager.java

```diff
@@ -0,0 +1,116 @@
+package com.example.sam.bucketlist.api;
+
+import android.app.Activity;
+import android.content.Context;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+
+import com.example.sam.bucketlist.R;
+import com.example.sam.bucketlist.models.BucketListFields;
+import com.example.sam.bucketlist.models.LoginFields;
+import com.example.sam.bucketlist.models.UserFields;
+import com.example.sam.bucketlist.service.UserClient;
+import com.example.sam.bucketlist.views.bucketlists.BucketListAdapter;
+
+import org.json.JSONException;
+
+import java.util.ArrayList;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Class APIManager Application Operations
+ */
+
+public class APIManager {
+
+    ArrayList<BucketListFields> bucketListData = new ArrayList<>();
+
+
+    Retrofit.Builder builder = new Retrofit.Builder()
+            .baseUrl("http://10.0.2.2:5000/api/v1/")
+            .addConverterFactory(GsonConverterFactory.create());
+
+    Retrofit retrofit = builder.build();
+    UserFields userFields = new UserFields();
+
+
+
+    final UserClient userClient = retrofit.create(UserClient.class);
+
+    public APIManager() {
+
+    }
+
+    public APIManager(String userName, String password) {
+
+        this.userFields.setUserName(userName);
+        this.userFields.setPassword(password);
+    }
+
+
+    public boolean login(Callback<UserFields> callback) {
+
+        LoginFields loginObj = new LoginFields(userFields.getUserName(), userFields.getPassword());
+        Call<UserFields> call = userClient.login(loginObj);
+        call.enqueue(callback);
+
+        return false;
+    }
+
+    public void getBucketLists(String Token, final Context context) throws JSONException {
+
+
+        String tokenHeader = "Bearer " + Token;
+        Call<ArrayList<BucketListFields>> call = userClient.getBucketlist(tokenHeader);
+
+
+        call.enqueue(new Callback<ArrayList<BucketListFields>>() {
+
+            @Override
+            public void onResponse(Call<ArrayList<BucketListFields>> call,
+                                   Response<ArrayList<BucketListFields>> response) {
+
+
+                try {
+                    bucketListData = response.body();
+
+                    RecyclerView recyclerView = ((Activity)context)
```

Comment:
You could pass the `Callback<ArrayList<BucketListFields>>` object as a parameter into the `getBucketLists` method. That way the `APIManager` class will retain its single responsibility of making API calls, and all the view modification will happen on the side of the activity.